tutorial/write-custom-widget.md

```diff
@@ -107,3 +107,178 @@ Take care not to keep any important state in your renderer - animation tickers
+func (w *MyWidget) MinSize() fyne.Size {
+	return w.minSize
+}
+func (w *MyWidget) Resize(s fyne.Size) { // See note below
+	w.BaseWidget.Resize(s)
+}
+```
+
+Note : It is not necessary to override the Resize function (shown above) but if you need to change the Resize behaviour of your widget and you add a ```Resize(s fyne.Size)``` function then you **must** include the ```w.BaseWidget.Resize(s)```. in the code.
+
+The final function to add is the ```CreateRenderer() fyne.WidgetRenderer``` function:
+
+```go
+func (w *MyWidget) CreateRenderer() fyne.WidgetRenderer {
+	return newMyWidgetRenderer(w.minSize, w.text)
+}
+```
+
+**Warning**: Do not keep a reference to the widget renderer in the widget (or any other part of your application). The fyne application caches the renderer and may destroy it at any time.
+
+The fyne application will call ```CreateRenderer()``` again if it requires a new renderer instance.
+
+It is ok for the widget renderer to refer to the widget but not the other way round.
+
+The widget is now almost complete except for the ```newMyWidgetRenderer()``` function which creates your new widget renderer. 
+
+#### Create a Widget Renderer
+
+The widget renderer is how we tell the fyne application what to draw and how to lay it out.
+
+The widget renderer has it's own state and functions (see ```fyne.WidgetRenderer``` interface) but this should only be concerned with drawing (rendering) the widget.
+
+The renderer is responsible for:
+
+1) The objects that are to be drawn. These are returned by the ```Objects() []fyne.CanvasObject``` function. Notice that you can return any object as long as it implements the ```fyne.CanvasObject``` interface.
+2) Each object returned by the Objects() function will need to be created by the widget renderer and initialised before returning.
+3) The fyne application will then call the renderers Layout() function. This is where the size and position of each object is detirmined.
+
+Our example renderer could be stored in the following struct:
+
+```
+type myWidgetRenderer struct {
+	minSize fyne.Size // The size of the widget
+	rect    *canvas.Rectangle
+	text    *canvas.Text
+}
+```
+
+As you can see it is about canvas objects that the fyne application will render. A lower case name is used as the renderer code is private and only created by the Widget. 
+
+The constructor is the place to pass in properties like minimum size, text values and other data.
+
+```go
+func newMyWidgetRenderer(size fyne.Size, text string) *myWidgetRenderer {
+	return &myWidgetRenderer{
+		minSize: size,  // For the MinSize() function
+		rect:    canvas.NewRectangle(theme.BackgroundColor()),
+		text:    canvas.NewText(text, theme.ForegroundColor()),
+	}
+}
+```
+
+Notice that the size and position of the canvas objects are not defined here, only colours, text and other properties. It might be appropriate to pass in a reference to the widget instead of lots of different parameters. This is left to your design requirements.
```

Comment:
I think we should recommend passing the object parameter - as mentioned above the renderer will show stale data otherwise.